src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -264,17 +264,50 @@ private void WriteDeltaResource(object graph, ODataWriter writer, ODataSerialize
                     }
                     else
```

Comment:
@ElizabethOkerio I'd say instead of repeating the line for retrieving the `EdmProperty` in both this `else` block and the `if` block above it, you move it right where the `foreach` block starts. You could also use the null-conditional operator to cater for the unlikely scenario of `entityType` being null. Something like this would be neater
```
                    IEdmProperty edmProperty = entityType?.Properties()
                        .FirstOrDefault(p => p.Name.Equals(prop.Key));

                    if (prop.Value != null
                        && (prop.Value is DynamicTypeWrapper || (prop.Value is IEnumerable<DynamicTypeWrapper>)))
                    {
                        if (edmProperty != null)
                        {
                            dynamicTypeProperties.Add(edmProperty, prop.Value);
                        }
                    }
                    else
                    {
                        object propertyValue;

                        if (prop.Value == null)
                        {
                            propertyValue = new ODataNullValue();
                        }
                        else
                        {
                            if (edmProperty != null)
                            {
                                propertyValue = ODataPrimitiveSerializer.ConvertPrimitiveValue(prop.Value, edmProperty.Type.AsPrimitive());
                            }
                            else
                            {
                                propertyValue = prop.Value;
                            }
                        }

                        var property = new ODataProperty
                        {
                            Name = prop.Key,
                            Value = propertyValue
                        };

                        properties.Add(property);
                    }
                }
```
You should also write tests to cover the scenario described in the issue. In addition, link the related issues (some dating back to 2017) to your PR such that they're closed when you merge your PR.

Reply:
Well Noted. Thanks